controllers/post-controller.go

```diff
@@ -0,0 +1,65 @@
+package controllers
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"go.mongodb.org/mongo-driver/bson/primitive"
+
+	"github.com/julienschmidt/httprouter"
+
+	"github.com/ulumuri/rhododendron/models"
+)
+
+type Env struct {
+	DB models.Requester
+}
+
+func (env *Env) InsertPost(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	defer r.Body.Close()
```

Comment:
You don't need to close the request body.

https://golang.org/pkg/net/http/#Request:
>
    // For server requests, the Request Body is always non-nil
    // but will return EOF immediately when no body is present.
    // The Server will close the request body. The ServeHTTP
    // Handler does not need to.

Reply:
Ack.